### ISSUE TITLE
Tuning weight decay

### ISSUE THREAD

**Issue by gurizab** [pomonam/Self-Tuning-Networks#5]
Hi, thanks a lot for making the code public!

I was trying to reproduce the results of tuning the weight decay on the linear regression toy problem.
While using batches of hyperparameters for e.g. dropout, color jitter, etc. makes sense, how do you aggregate the results of having a batch of different weight decay values into one regularization value?

Thanks in advance!
Guri

**Comment by pomonam** [comment]
Hi, thanks for your interest in the codebase! When computing the final loss, you can perform the following:
```
final_loss = torch.mean(loss + batch_wd * norm), 
```
where `batch_wd` has a dimension `batch_size x 1` and `loss` also has a dimension `batch_size x 1`. In this way, you can aggregate a batch of different wds. I hope this helps!

**Comment by gurizab** [comment]
Thanks a lot for the clarification!
